### PR TITLE
Make configen compatible with PEP 585 annotations

### PR DESCRIPTION
**Pull request by tmke8**

### CODE REVIEW

File: tests/test_modules/generated.py [hunk 1]
```diff
@@ -93,28 +94,33 @@ class WithUntypedStringDefaultConf:
...
 @dataclass
 class DictValuesConf:
     _target_: str = "tests.test_modules.DictValues"
     dct: Dict[str, str] = MISSING
+    dct2: dict[str, str] = MISSING
     enum_key: Dict[Color, str] = MISSING
     dataclass_val: Dict[str, User] = MISSING
     passthrough_dict: Any = MISSING  # Dict[str, LibraryClass]
     def_value: Dict[str, str] = field(default_factory=lambda: {})
+    def_value2: dict[str, str] = field(default_factory=lambda: {})
```
Comment (MylesBartlett) [inline tests/test_modules/generated.py:114]:
```suggestion
    def_value2: dict[str, str] = field(default_factory=lambda: {})
```
Is a lambda necessary here? Can't one simply use `dict`?

Comment (tmke8) [reply tests/test_modules/generated.py:114]: yes, `dict` should work

Comment (tmke8) [reply tests/test_modules/generated.py:114]: ah, but this is generated code, so I'd rather not mess with it

**Comment by tmke8** [comment]
Why did the CI pass when it was still using Python 3.8 while failing now with 3.9!?

**Review by MylesBartlett — APPROVED** [review]

**Pull request merged**